### PR TITLE
Clean goes xrs tests

### LINKED ISSUE #6457: GOES-XRS tests need tidying and updating
### Provide a general description of the issue or problem.

The `XRSTimeSeries` supports a range of files and the testing suite to test these is a bit all over the place and doesn't fully test all possible file paths. 

The files that XRSTimeSeries can read are:

1. GOES-R XRS 1s flux (netCDF files)
      - GOES 16 files
      - GOES 17 files
2. GOES-R XRS 1m averages (netCDF files)
      - GOES 16 files
      - GOES 17 files
3. GOES XRS 2s fluxes (netCDF files)
      - GOES 13 files
      - GOES 14 files
      - GOES 15 files
4. GOES XRS 1m averages (netCDF files)
      - GOES 13 files
      - GOES 14 files
      - GOES 15 files
5. GOES XRS 2s fluxes (FITS files)
      - GOES 1-15 files

So basically 5 different file types that need to be tested. Most are, but I think a tidier way of doing this would be good.

### PR DESCRIPTION
**Pull request by aritrasinha108**
## PR Description
closes #6457 
This PR inculcates all the GOES files by parametrizing them so that all of them can be tested at once for all the tests.

### CODE REVIEW

**Comment by aritrasinha108** [comment]
@nabobalis Could you see if the changes look fine?

File: sunpy/timeseries/sources/tests/test_goes.py [hunk 1]
```diff
@@ -43,32 +43,53 @@ def test_implicit_goes_satno(goes_nc_files, sat_no):
...
-    assert ts_goes.time[0].strftime("%Y-%m-%d %H:%M:%S.%f") == '2013-10-28 00:00:01.385'
-
+    ts_goes = sunpy.timeseries.TimeSeries(goes_nc_files, source="XRS")
+    assert ts_goes.time[0].strftime("%Y-%m-%d %H:%M:%S.%f") in ['2013-10-28 00:00:01.385', '2012-05-31 23:59:59.089', '2017-09-01 00:00:00.631', '2019-01-02 00:00:00.000' ]
 
-def test_goes_netcdf_time_parsing17():
+@pytest.mark.parametrize("goes_nc_files",
+                          [goes17_filepath_nc,
+                          goes16_1m_avg_filepath])
+def test_goes_netcdf_time_parsing17(goes_nc_files):
```
Comment (nabobalis) [inline sunpy/timeseries/sources/tests/test_goes.py:88]: If the time stamp also varies, we should parametrize that.

Comment (aritrasinha108) [reply sunpy/timeseries/sources/tests/test_goes.py]: Parametrized the results too now.

**Comment by nabobalis** [comment]
The precommit needs fixing but LGTM

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @aritrasinha108

**Pull request merged**